moveit_planners/chomp/chomp_motion_planner/src/chomp_trajectory.cpp

```diff
@@ -190,68 +190,38 @@ void ChompTrajectory::fillInMinJerk()
 
 bool ChompTrajectory::fillInFromTrajectory(const robot_trajectory::RobotTrajectory& trajectory)
 {
-  // get the default number of points in the CHOMP trajectory
-  const size_t num_chomp_trajectory_points = (*this).getNumPoints();
-  // get the number of points in the input trajectory
-  const size_t num_input_points = trajectory.getWayPointCount();
-
   // check if input trajectory has less than two states (start and goal), function returns false if condition is true
-  if (num_input_points < 2)
+  if (trajectory.getWayPointCount() < 2)
     return false;
 
-  // variables for populating the CHOMP trajectory with correct number of trajectory points
-  const unsigned int repeated_factor = num_chomp_trajectory_points / num_input_points;
-  const unsigned int repeated_balance_factor = num_chomp_trajectory_points % num_input_points;
+  const size_t max_output_index = (*this).getNumPoints() - 1;
```

Comment:
```suggestion
  const size_t max_output_index = getNumPoints() - 1;
```

Lately I've been having doubts of the usefulness of `this` :) Feel free to ignore though